src/device-manager/WeaveDeviceManager.cpp

```diff
@@ -2921,7 +2950,14 @@ void WeaveDeviceManager::ClearRequestState()
 
 void WeaveDeviceManager::ClearOpState()
 {
+    if (mCurReqMsgRetained != NULL)
+    {
+        PacketBuffer::Free(mCurReqMsgRetained);
+        mCurReqMsgRetained = NULL;
+    }
```

Comment:
I intentionally left it without ifdef because this logic is general and might be reused in the future by other profiles that introduce new version of a message.